network/src/node.rs

```diff
@@ -336,7 +340,10 @@ impl Node {
         }
 
         // The node can already be at some non-zero height.
-        metrics::gauge!(misc::BLOCK_HEIGHT, self.storage.canon().await?.block_height as f64);
+        if let Some(storage) = &self.storage {
```

Comment:
this is fine for now, but I'm guessing that peering-specific nodes that _will_ have storage for the storage of lists of peers will have to be special-cased here too; alternatively, this could already be changed to something like:
```
if self.sync.is_some() {
    <use expect_storage()>
}
```
as storage will always be a given if there's a sync layer, and chain-specific metrics primarily depend on the sync layer being available